### ISSUE TITLE
output is  !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

### ISSUE THREAD

**Issue by BeSkyer** [volcengine/verl#721]
While training with GRPO, I encountered an issue where the model output abruptly degrades into a sequence like !!!!!!!!!!!!!!!. Just before this degradation, the kl_loss becomes NaN. Upon further investigation, I found that old_log_prob is turning into NaN, which appears to be the root cause.

Additionally, the occurrence of this issue is inconsistent across training steps, indicating that it likely isn’t related to the dataset (I'm using GSM8K). I’ve also noticed similar reports from others using GRPO or REINFORCE++.

Are there any plans to address this instability in future updates?

step:59 - global_seqlen/min:16268.000 - global_seqlen/max:20607.000 - global_seqlen/minmax_diff:4339.000 - global_seqlen/balanced_min:17865.000 - global_seqlen/balanced_max:19434.000 - global_seqlen/mean:18257.750 - actor/kl_loss:nan - actor/kl_coef:0.001 - actor/entropy_loss:nan - actor/pg_loss:nan - actor/pg_clipfrac:0.002 - actor/ppo_kl:nan - actor/grad_norm:nan - perf/mfu/actor:0.042 - perf/max_memory_allocated_gb:27.050 - perf/max_memory_reserved_gb:28.637 - perf/cpu_memory_used_gb:43.031 -

**Comment by Enoch202** [comment]
I encountered the same problem when training Qwen with GRPO or reinforce++. 
The crash during training was very sudden: the previous step was normal, but in the next step, the loss became NaN.

**Comment by vermouth1992** [comment]
Could you provide a script that can reproduce?

**Comment by Reason-Wang** [comment]
Same problem here. Has anyone investigated this problem? It happened on my side on math problems with the dataset from [Open-Reasoner-Zero](https://github.com/Open-Reasoner-Zero/Open-Reasoner-Zero). It can happen on any step during training.

**Comment by yushuiwx** [comment]
Hi, same problem, @BeSkyer May I ask how you managed to solve it, please?

**Comment by BeSkyer** [comment]
> Hi, same problem, [@BeSkyer](https://github.com/BeSkyer) May I ask how you managed to solve it, please?
Sorry for the late reply. I spent a long time checking the code but still couldn't solve the issue. In the end, I set KL_loss to False, and that prevented the NaN problem from occurring.

**Comment by chuangzhidan** [comment]
> > Hi, same problem, [@BeSkyer](https://github.com/BeSkyer) May I ask how you managed to solve it, please?
> 
> Sorry for the late reply. I spent a long time checking the code but still couldn't solve the issue. In the end, I set KL_loss to False, and that prevented the NaN problem from occurring.

 it's not working for me
    actor_rollout_ref.actor.use_kl_loss=False \
    actor_rollout_ref.actor.kl_loss_coef=0.001 \
    actor_rollout_ref.actor.kl_loss_type=low_var_kl \

**Comment by scris** [comment]
use_kl_loss=Falsem so kl_coef works, not kl_loss_coef.

**Comment by zhenghaoxu-gatech** [comment]
> use_kl_loss=False, so kl_coef works, not kl_loss_coef.

I got the same issue with kl_coef=0.0, use_kl_loss=False, kl_loss_coef=0.0, and no metric is NaN when output collapses to '!'

**Comment by Neroxsh** [comment]
The same problem, has anyone solved it?
